### PR TITLE
homebrew configs for nightly builds

### PR DESCRIPTION
**Pull request by wchengru**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: bottle-configs/aws-sam-cli-nightly.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "version": "1.21.1.dev202101010000",
```
Comment (sriram-mv) [inline bottle-configs/aws-sam-cli-nightly.json:2]: will this be automated by a bot?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: bottle-configs/aws-sam-cli-nightly.json (hunk 1 above)
Comment (wchengru) [reply bottle-configs/aws-sam-cli-nightly.json:2]: yes, it is a placeholder now. The sam-cli bot will update it once source code is tagged.

File: Formula/aws-sam-cli-nightly.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  desc "AWS SAM CLI 🐿 is a tool for local development and testing of Serverless applications"
+  homepage "https://github.com/aws/aws-sam-cli/"
+  url config_provider.url()
+  sha256 config_provider.sha256
+  head "https://github.com/aws/aws-sam-cli.git", :branch => "nightly-builds"
+
+  bottle do
+    root_url config_provider.root_url()
+    cellar :any_skip_relocation
+    sha256 config_provider.sierra_hash() => :sierra
```
Comment (hoffa) [inline Formula/aws-sam-cli-nightly.rb:18]: How long are we planning on keeping `sierra`?

File: Formula/aws-sam-cli-nightly.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+require_relative '../ConfigProvider/config_provider'
+
+class AwsSamCliNightly < Formula
+  include Language::Python::Virtualenv
+
+  config_provider = ConfigProvider.new('aws-sam-cli-nightly')
+
+  desc "AWS SAM CLI 🐿 is a tool for local development and testing of Serverless applications"
```
Comment (mndeveci) [inline Formula/aws-sam-cli-nightly.rb:9]: nit: Do we need to mention nightly in the description?

File: Formula/aws-sam-cli-nightly.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 config_provider.linux_hash() => :x86_64_linux
+  end
+
+  depends_on "python@3.8"
+
+  def install
+    venv = virtualenv_create(libexec, "python3.8")
+    system libexec/"bin/pip", "install", "--upgrade", "pip"
+    system libexec/"bin/pip", "install", "-v", "--pre", "--ignore-installed", buildpath
+    system libexec/"bin/pip", "uninstall", "-y", "aws-sam-cli"
```
Comment (mndeveci) [inline Formula/aws-sam-cli-nightly.rb:28]: Should this line be `system libexec/"bin/pip", "uninstall", "-y", "aws-sam-cli-nightly"`?

Comment (wchengru) [reply Formula/aws-sam-cli-nightly.rb:28]: I think here we do the uninstallation to make sure there is no conflict with aws-sam-cli. I saw aws-sam-cli-rc also uninstalled the main package: https://github.com/aws/homebrew-tap/blob/f8aeb5034a95beb70f8cc3e938c9276fd813bad9/Formula/aws-sam-cli-rc.rb#L30

Comment (mndeveci) [reply Formula/aws-sam-cli-nightly.rb:28]: But this one is not conflicting with `aws-sam-cli` anymore, can we verify if we still need this?

Comment (wchengru) [reply Formula/aws-sam-cli-nightly.rb]: I guess we are still sharing pypi with normal sam cli.
Test removed this line, the symlink doesn't create.

File: Formula/aws-sam-cli-nightly.rb (hunk 3 above)
Comment (wchengru) [reply Formula/aws-sam-cli-nightly.rb]: Updated.

**Pull request merged**